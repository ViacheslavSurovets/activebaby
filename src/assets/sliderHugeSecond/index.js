import sliderHugeSecond_1 from './sliderHugeSecond_1.jpg';
import sliderHugeSecond_2 from './sliderHugeSecond_2.jpg';
import sliderHugeSecond_3 from './sliderHugeSecond_3.jpg';
import sliderHugeSecond_4 from './sliderHugeSecond_4.jpg';

export {
  sliderHugeSecond_1,
  sliderHugeSecond_2,
  sliderHugeSecond_3,
  sliderHugeSecond_4
};
