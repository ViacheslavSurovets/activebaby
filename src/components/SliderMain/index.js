export { default as SliderMain} from './SliderMain.jsx';
