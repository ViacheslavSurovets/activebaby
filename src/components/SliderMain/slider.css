.tu div  {
    animation: popUp 6s ease 1;
    opacity: 1;
}

@keyframes popUp {
    from {
        opacity: 0;
    }
    to {
        transform: translateY(3rem);
        opacity: 1;
    }
    15% {
        opacity: 1;
    }
    25% {
        transform: translateY(3rem);
    }
}