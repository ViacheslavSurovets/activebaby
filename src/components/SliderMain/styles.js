import styled, { keyframes } from 'styled-components';
import { Link } from 'react-router-dom';

const popUpSliderMediaHuge = '@media (min-width: 400px)';
const popUpSliderMediaLow = '@media (max-width: 401px)';

const keyframePopUp = keyframes`
    from {
        opacity: 0;
    }
    to {
        transform: translateY(3rem);
        opacity: 1;
    }
    15% {
        opacity: 1;
    }
    25% {
        transform: translateY(3rem);
    }
`;

export const Slide = styled ( Link )`
  display: flex;
  background-image: url(${ props => props.background }) ;
  background-size:  cover;
  background-repeat: no-repeat;
  text-decoration: none;
  ${popUpSliderMediaLow}{
    flex-direction: column;
    width: 100vw;
    height: 140vw;

    &.js-slider-main > div { 
      opacity: 1;
    }
  }
  ${ popUpSliderMediaHuge }{
    justify-content: flex-end;
    align-items: baseline;
    width: 100vw;
    height: 75vw;
    &.js-slider-main > div { 
      animation: ${ keyframePopUp } 6s  ease 1;
      animation-fill-mode: forwards;
      opacity: 1;
    }
  }
`;

export const PopUp = styled.div`
  display: flex;
  align-items: center;
  flex-direction: column;
  justify-content: space-around;
  flex-wrap: wrap;
  ${ popUpSliderMediaLow }{
    background: ${ ( { theme } ) => theme.main.sliderPopUp.popUpBackgroundMediaLow };
    color: ${ ( { theme } ) => theme.main.sliderPopUp.popUpColorMediaLow };
    position: static;
    height: 70vw;
  };
  ${popUpSliderMediaHuge}{
    margin-top: 5vh;
    margin-right: 2vw;
    width: 20rem;
    height: 25rem;
    opacity: 0;
    min-width: 20rem;
    min-height: 25rem;
    background: ${ ( { theme } ) => theme.main.sliderPopUp.popUpBackgroundMediaHuge };
    color: ${ ( { theme } ) => theme.main.sliderPopUp.popUpColorMediaHuge };
  };
`;


export const PopUpText = styled.span`
  font-size: 1.4rem;
`;

export const PopUpTitle = styled.span`
  font-size: 2.5rem;
`;

export const PopUpButton = styled.span`
  display: flex;
  justify-content: center;
  align-items: center;
  height: 4rem;
  width: 50%;
  ${ popUpSliderMediaHuge }{
    width: 50%;
    font-size: 2rem;
    background-color: ${ ( { theme } ) => theme.main.sliderPopUp.popUpButtonBackgroundMediaHuge };
    color : ${ ( { theme } ) => theme.main.sliderPopUp.popUpButtonColorMediaHuge }
  };
  ${ popUpSliderMediaLow }{
    font-size: 2rem;
    background-color: ${ ( { theme } ) => theme.main.sliderPopUp.popUpButtonBackgroundMediaLow };
    color : ${ ( { theme } ) => theme.main.sliderPopUp.popUpButtonColorMediaLow }
  };
`;

export const Img = styled.img`
  display: none;
  ${popUpSliderMediaLow}{
    display: inherit;
    width: 100vw;
    height: 70vw;
  } 
`;
