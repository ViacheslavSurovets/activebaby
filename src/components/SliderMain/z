import { cristiano, chantal, adamex } from '@assets/sliderMain';

export default ( content = 'Slide' ) => {

  const slideData = [
    {
      title: 'Adamex',
      url: adamex,
      popUp: {
        info: 'info',
        title: 'Title',
        text: 'text',
        button: 'button'
      }
    },
    {
      title: 'Cristiano',
      url: cristiano,
      popUp: {
        info: 'ttt',
        title: 'ageag',
        text: 'text',
        button: 'button'
      }
    },
    {
      title: 'Chantal',
      url: chantal,
      popUp: {
        info: 'ttt',
        title: 'ageag',
        text: 'text',
        button: 'button'
      }
    }
  ];


  return  slideData.map ( ( item, idx ) => ({
    idx: idx + 1,
    content: `${ content } #${ idx + 1 }`,
    url : item.url,
    title: item.title,
    popUp: {
      info: item.popUp.info,
      title: item.popUp.title,
      text: item.popUp.text,
      button: item.popUp.button
    }
  }) );

};
