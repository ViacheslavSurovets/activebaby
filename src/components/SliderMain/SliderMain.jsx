import React, { useCallback } from 'react';
import Swiper from '../../core/swiper';
import { PopUpButton, Slide, PopUpTitle, PopUpText, PopUp, Img } from './styles';
import { SliderWrapper } from '@commonStyles';
import PropTypes from 'prop-types';

const SliderMain = ( { items } ) => {

  const params = {
    ContainerEl: 'section',
    slideActiveClass: 'js-slider-main',
    slidesPerView: 1,
    navigation: {
      nextEl: '.swiper-button-next',
      prevEl: '.swiper-button-prev'
    },
    autoplay: {
      delay: 5000
    }
  };

  const renderItem = useCallback (
    ( { idx, imageUrl, title, popUpData, path } ) => (
      <Slide to={path} key={ idx } background={ imageUrl }>
        <Img src={ imageUrl } alt={ title } />
        <PopUp>
          <PopUpText>{ popUpData.info }</PopUpText>
          <PopUpTitle>{ popUpData.title }</PopUpTitle>
          <PopUpText>{ popUpData.text } </PopUpText>
          <PopUpButton>button</PopUpButton>
        </PopUp>
      </Slide>
    ),
    []
  );

  return (
    <SliderWrapper>
      <Swiper params={ params }>
        { items.map ( renderItem ) }
      </Swiper>
    </SliderWrapper>
  );
};

export default SliderMain;

SliderMain.propTypes = {
  items: PropTypes.array.isRequired
};
