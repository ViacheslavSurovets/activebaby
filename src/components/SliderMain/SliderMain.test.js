import React from 'react';
import { shallow } from 'enzyme';
import SliderMain from './SliderMain';
import { sliderHugeSecond_1 } from '@assets/sliderHugeSecond';


describe ( 'SliderMain', () => {
  const mockItems = [
    {
      idx: 1,
      title: 'Adamex',
      imageUrl: sliderHugeSecond_1,
      path: '#',
      popUpData: {
        info: 'info',
        title: 'Title',
        text: 'text',
        button: 'button'
      }
    }
  ];
  it ( 'SliderMain Snapshot', () => {
    expect ( shallow ( <SliderMain items={ mockItems } /> ) ).toMatchSnapshot ();
  } );
} );
