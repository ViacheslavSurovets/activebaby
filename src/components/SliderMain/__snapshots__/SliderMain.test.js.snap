// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SliderMain SliderMain Snapshot 1`] = `ShallowWrapper {}`;
